packages/presentation/src/preview/PreviewHeader.tsx

```diff
@@ -101,7 +101,7 @@ const PreviewHeaderDefault: FunctionComponent<Omit<PreviewHeaderProps, 'renderDe
     return targetOrigin === location.origin ? '' : targetOrigin
   }, [targetOrigin])
 
-  const handleRefresh = useCallback(() => {
+  const handleRefresh = () => {
```

Comment:
In this case I removed `useCallback` as the React Compiler was bailing out, out of an abundance of caution. 
As `handleRefresh` is only used as an `onClick` prop it's not a big deal exactly how it's memoized, [I've manually verified it](https://playground.react.dev/#N4Igzg9grgTgxgUxALhASwLYAcIwC4AEwAOgHYEEDCAFgnANYYCGM9AknBKQDRkU0IAbjC4ARCAHdSHLr3IFRCMPTwQsMnnwIBRACZo8GuRQCyEAEZoANgkWC0iI1oAKTUgisAZBADNDnTXlnKHMrNDBqJ3kAJV8YJUiAuQBfAh8RDAIAcgABMDcDAE8AegcuMCyyTBx8IgIJA2poPABlOnjWhBY4alcYJgwwAlT0iEzc-NIi4qx4+wQJAFpYK0WwdoQ8Yoa8Jqg8NY2D9e7qRawWAYqq7FxCEnkAIQgAD2MCR-3VQP4WXXeAGI2N5aAASEDw9AQhTA7xMCFIUDhCKgnzw32RiLYeAQGHeLTwTAY+J2PXeABUEC88BSIBArHg0Fh3nhClgEB8vlxyVwECk0hlsnkCqzilA0JUprdag8KHB4kwcdobBgETStFB1pQmFYrOYifR3pqEPCMBAWWyOZQxjh3KQ8OTLRb2QQAVBSHBGVxrbc7er5KyXbEiXgAHIQXR8rSBjmxHwAeXMACs6P6RoKsgrPZLqnc6sbyf1SGArIq0FxhgKxtlJkUc9L7gRZko1WWuJ4IHAdQhQwMlBdEJXRuMAHQj0oARgAHKR6zVGwBBSjktjx0MAfTYAOiC5M2nX0W0nnjC9EQ4zY+K8V0UEQMDAM3i63tbdIsRvd7neZjdVBXUjMDxlgXrFtwBDOE+rYgc4CD3uynpoIIHLptWWSXjG1xSvOdRAQiEFCGgCxot856oeOuGkPh8wSMRXBfrUP7AOBcyERIEFqEMKGjsUVGsfRja8QsHZdiBbCkFg+ykdxgkSMJr5iRJeD8XULTUCwCAyfCiJSdk46qepmkopKZABGAhDONo0QtBZy5sAAavuK7kp4+4ANLaAAmsgBCxJwMC6AAPPhz6EtBsFgPBjJIWBZkwGgpAAOYAHwEAAvEQWjNtRoj9H4YDeVkWWsYs6R9iO7JwamiEIOVLELDlTB5SOjJ4DYWTvBJoThLQugFUVCwlf0qrleFkXVeVIRhBECC6M1BhtSkZAmeU5mWdZ2i2Q567kmu2gtN5vm4IFwVQeWlGjVV0Wcui3K8il6Wyk2dUSA1eUFV2+xne1mWTd1M19RAYAGNV32kMkS0eit4FrTZK5bWwlBrvtPl0EdQWQS+YWVQhV0+raaqOuy90ZfI-UvbleD5To+j+LIP1ddNvXgb9ERRODpBkFS2HxTiMA+ESHIyX+TAAexWBDFSOKkLoQwyWLQyPWgpWqnGB2+ImKaegFoLkiYnhbkNCDKriapJVo8TS7Bij81ADLeQAFLMHHeUL-6wfLACUaUpcGnrhpGZDs8txbmc9wsAdbTC23g3luh6IF47y9oBfGGAGOjBELOH7siOLYGZgiEe+FHDJZElxP25luewmQXupSlj2mfcWhym4+nxDJC5wIgYA1-IrekDyCUJTY7cac9Xc933FADwAqusY+6J33dKNPM-6BFipki3BBK4b7wUHvfZxgfu9TIyOqzzAVin1YEAi+F1qkO4CF0-3BCkEw9gJYquDaJ-oQZqny4K4XYNA3AJSjO-LgcYnzUGAeySiagoDMh3hAJC18mAwifi-ECp8KoRUulAmeT1M4SCvjfHe6w8AwWxlFYhM9qF2VYvOU+hIYCQLwPGOKCV4psIgMPGwvZv6-xgPwwRCB4wYNLIUU+1FWE73sGAKAOo9DA0St5Ji6DYIyLAP-JggDdDDHeKkdKTtxYQzlFDJihBTEEALEWEsr5HYY1CmdOSQi+yb0QB7SxBAm4EFUBI5hCx5wtDQAALw5OlY02pdT6gYJXd+9s64pSYSwu49t5F5lSrk7IkZlCqCwFkAgAB+bIZpLBtQIAVApKg1BZA9qfAA2uk0JdwwLZPwAAXXeL4jm8gAlkzkiBbhaBeHkBiesU0EB7YpO9iTEhHRYDkHYZwsZEy0p5LviJM6I5cDjPimU7IJTvJrM2Bs+KWhkhgWaecrhPD4rdP6VoAJalpY2FgQkNKBB5n10WTPGBcQEhzNSQCkhStfkAEIj4q18COOAsALZ4C9o9EhFBlkwAGei4YO8KAbwuHgHo9sbGWm8kuFca5Nzbl3PuQ8x5TzDH6Ti4oxRXTunoIUeoWDAkQAIPEO+IsCBuF3srDkOw9iEERlZXevcoBKDxQQVlBAlBhHtIsDeBibCLHcNSRYaqOSkAgIcKwPhFhMF7uM7F6LlXxB8AipFaoRxgHgD8u1DqYDIudfARVGtUwjgtUDBKpB7awoQHGD1yKwJMRdXAbyAADAAJMAe5lywbJrJhQggAAfbN2RihZGSPGplO9kjMtxdaoZz0RlnWiM0aJ9jpm4lmX8huO8q1kNntETwPz3ASAIF2zwLjO3XxzXmrIBawKpseaQctViQ51EJdQT+qoYpdHgNQSs6UJXNDaPKTYbRTh9CuMO6ig6Xnv0xeQJNwAl0roQMkZNJwN1FuubczN18p0sHWTO55fir2-J3gFM278KApoESPHsX9xmiIIAAMjg4B0DM8ArPBeGkYEqViAgCNe4bDBBmAcPitEcZ1A8CpWABOVIFxdD6EShRqjIGcUkICjyekjIsCKpxQUZgOIuPooCFLcjwBWOSwIEDKJDHkhJRTfbQqz1BplU-iI1QMBmoQe1eYLkpB1PsaZI05IAViiUmpEldmzH0X83iQaZwpZECqntGACjzSsjmAhKoDAax2FKW6eZizM8aN0YShRgATH5-zTY7Mm3tFhkAbmboYGw-xkh84dTJYIExiLBBUPafSyQlgaAmD6oMR4CjeA5Nk0U8NZTMHVPqYkYsLTN0dMFaK6WcwHgDN5Znj4LgrRIkICk91w+AQKOuHcF4XwtMwbDYIxGQb2HAEzSS8hiz3owgMDKxpqDKncDhay4F+KwXgBhdm+sGwnoZoUZqz-VT+jDH7Ys8UTLFmjNsYZEyF76KjNoa+xQD2QdkOsbpB9zjq3hVSkVAgfjgm1QUfSwFIEVJhVhGDbFxA9pYL4bMoUGwFGmISGoAYBALQBwIAKkaiQ-RinDGk911DrwmwiyC1Jv7-nRPUnEwN1ns2wPlfk2QqrNVtGYMKPVyDjXtO6dB+1cFWWSGw+pN5EXuj7s2CMeUrImrAGnOyAiLVCBQby5IWWx77PjOSzZzin7rwrffbQ0z2jR2PI8-B9b8EkJoRDChDCZzWQFyl18+J1kePgBMUIxMnkWBvKLAACxgQj-FZ4CWY+x9p0qu3LHii-YR8UJHLw7dm4oFZvUNmosOcpn7+LHmvMsB80Xx3LPKMN6wOXuHi33NjBWxZ1LVhFV24Ctqfy6WLWxfax4bv-mfCYew7h6HIB0uJ7fKR4TVH0uHfo8AAAzA3igOPQ9ov8wav8K-vIAAZT4Wbc-5WC0QRZoE1N5AAnC-y-zHjQwDaBdmO2Q59G-88kLvryu4BRirlgnogAursclkHgI4hcMirru6JGD4PFDNA3pntlvnijlaujmqFjiAAQD-FgBRjvhgRQAFPoIIMHrjoNmHgRt+vFKnrTmQShi0KSNQLzv4rQAwFdsAGATCGrmgZwd6O8pAlthIlIjolgkATitrurhRmGs6qFJqFsulFkIKkFiUrmgQPwWADglVBWFCnklkAELgjNIWrNs9vTsUJQSwQzgXrNhzoQBgPsLwVCroYIboKkBJrQVRvUAgCvrFqqPoFAIliACwSQrJgLtRELvstIlguLpplLjiNSF1m7t9hbqZtYTnukUZvngPsUEProGzm9iDhxiBoqvYRhlSFJvQURqQAABqu6vYyQ1pcAKT7Dr52poAvDw7pHkEO4b7HaMaOHvYcacE8ZQ6cGw4xaH7G7ZYmaEA+HNHzHorACKFmSKgqG5JqEaHmGcE4rlL86VZiojh2pwKS7NbS4caNIHHopnIVYKanGbF4Caiy5MSK4-4bHKGcQA53FvaW6cEyGWY6il4MC2YCwV5ObAAubV5jCLCFxZC+acFDGhbAkpZt4xYd4JaT7zG96zYREoa0TWqrHCpxRtYlZWBlaPGC6nHnEJCXHfABrknFYdZWBpGknF59bhKSbN53FlCkAUZfKswBDonopmiRixZLa6C4mrEaFHYKHPE-GqHZACr3yaFjqiqGxKFbFDA7GqnAoRBHYWH9HMbrYOD0AUbvK6CfKGnUBikpbM4KknYOkZ6OHGZlGfbZG27pZAEiEQK0EgKKjUDgKJQPrpYHITIUbDKdivhppAHKI+AoG9FzHMZVGonN6EkpyIIyTEnCE5nILEF8EFkSRYCukxm7JcBprRnVqxmjIzrlm1mVlvj1o1lkJtEtmuFilWGmk24OHpFAGCA6jyptnUQdl1pdn8Y9nW7Z4+kkkoZYE6g4HYYY68z4bT41Gz68j4ZEG1EZkjHg4BRaSoi5bpFNbfB9Hy45bNaza9b2g8m+FikCkkYJRkYUYCDCBiCSDSCimzYSkLZxY2DLYL6mkUAZmnagXiZk5omzYpHCblacEWRWSwz2SORsDORuSeTNKcF7H3j6GvyTLGGmGpjmHHIEJjRIQ1LZBkyvSUySikm9KzZ-GQXymb64V6FcC4JnQqkFzEUISJQlIIZallQvEqFGG7HqnmFilyG8HsX4UgQEDiXZB8U4i6Amny7TkAHpZoC6CxbkVEKLAOZQCyk4pGWXlZZHkojUEH70EvAADqOluw3kIWseZ+zBjhx52IuInBd5-WvJa+kFFA5pm2wAra4mmwtChCOMCANJ2UFMFQzFpJGZO+KJT46wult6F00VPFNF8V6lqx3wtBSF60m0jku0LQtUZCtFYAYphJ5Bi5qOgp2GG5Lwte+AO5TARZpB-xDuLVsWc+2GdV32ixXO-l0mwA+6UOxsFe9sxVKF8MiMoYFVuVjUlMAOAJWRgVKGs5-ZnJ2WBIBo1RKZfh3itBYWQ1LGI1yxze-hgR2GwRD+YRF1ax5Wc1G0cMaFGF647kHkLmK1eUSJiVe12WmReAz15BI1Lhqlo1j54NM8URJx2p+l0VlVcVq1YAzUksHJwNG1YN-xxQB1DAz1VRfVW5eG4RdxENkslNM8UNy2W1OK11AVwNjCIetBqZe1agRIRQ3kyN9COVz01VJS5SE4VFF+NNFAgBNNcN5BAgDAhG7AAQbpDN21ixxNO1Q1eRwIWZxQnlOIGAWZet3lW1vlD5Q2W1wVlpoVYK1CkVFFMVhULMPU2NeJTpm+KVW1zYvcvBfN1UOVTtUlnBhVY2MM71qF205VE0DMPUtVnBiOwI2BaOzVwIbVeAHVXVdOKtJNM+OG25FNKtVNnOTN41k1SoKoaos1odpVm4i1y1Adug61oN6tORe1AUhN9AR1tRp1aJcNThMNtREgARr55G91M0j1g1EtRAr1VdH1206FLk31WFjt0d5hzyrpWeatNNfddNRixdMtU9sVxUpxvtSEUdU0PUmNqRQNrdTd+N7dxNvVOdA1+dLNThk9O9k9xdn9bN+Ok9QVA4PNTYWV-N+py959pFItYtb+pJUtBdGWk9g+3BjALAitFYmlN9m9WdGtcdee2t7pRtBt1hx5GBQBOAOAGCv9ppkxOIZyMA8q0DAmK0-Q8UD5tD9D3WrekJaoBUcJGA-+EWvebDDCFmsDT2JRuDVIFRQODVy5IAq5+BndZN8+hBnVKx1uYxTII+kOfG6RMxwm9OV13OmZs2CNTx2pXSBw55XA1x+m0a0xfWks3kFjPFdSRSwtv+LAIgEguuWQPgts7JDDOKpu1hmDEWQBJeCS9AEJ9mao0JsJnenmmx+ASJpDbtx2EFAjmJI9cWCTJl6K+J4OGB153w3WrWrJpWpjtJ5jGS+AjJ1jZT4+7J19U+3JRjzNWWApFGzjoDrjDSxyZgVStgBEjgSt3kighSagGgwJ-5UpQFMpIF8ult4hkGISEgYSA2wJ4FQB6DGRGjWA4j+RfikRXYpAY8y8U88GiGSSr2MjSdcjeBMA65T9edKjRZfhQxTRxjuRZzz0x53WJzQ8kGPzZCk8q8FGAL22wL1EoLvcwJJz88JOak145zYLE1bgCLi8KLsLHDTZr4E5OIo5rE459awJ8UwMl818Ch58hWVgFCmzwD1U0ZDLSE2zBRhzyGzTRmmW-SgOASrsD8MAsc7oBFicfoAUWgqc6c-Los1c+cFsRcNspcKUiGWiwEZ0YApS3k2cgEar5QFaFc5iYAYKjcUM8rVsxc0cPysSoJkTcyhr3kkreAGc1E2r8scrhc5rirSkSUxrO8AGpdRs5d9os1YcbsMAkc0cYEhr5aNyBAzSf6laUMtAArPyhr+yurxYpSCKNoScad785SAb01FdabagIEGNnAvoao0aBAY4tUHEYEZr4bFrDIJa783kjbEbDILiHEF6GKmwKySGLGRRijud5NjezpYW4makugkgXdP9dBOAQMIEBUAqZYSEJSgBJRtzTV9zmOjzBB++7NBG8UJ+w95+7l4OwAybAEZuWtkjQGhRfw3LgcEMxQAAVG+wQDkDzLBJ-FYAQG+8UJzC8NhH4-HNxcaCdJjGdNK7BNc09M7AQI6866xK67KwaZbE2162XPBnUKW+q5qwQNq0BGW8YrXA7GCr7GGPNgCgEqK2qFa1qDawaKCgsofgBih1nGG0QLW4a6kJpbG80oawm+bP21ilQDm36C+6QCAMkEAA), it's fine.